### PR TITLE
Stanislav

### PR DESCRIPTION
**Pull request by Vilkoc**
almost finished

### CODE REVIEW

File: init.py [hunk 1]
```diff
@@ -10,15 +10,14 @@ class BasePage(unittest.TestCase):
...
         cls.browser = Methods(cls.driver, TIMEOUT)
         cls.header = Header(cls.driver)
 
-
-
     @classmethod
     def tearDownClass(cls):
-        cls.driver.quit()
+        pass
+        # cls.driver.quit()
```
Comment (TanyaKalistruk) [inline init.py]: code in comments

File: pages/auth_page.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    def click_sign_in(self):
+        self.click_element(self.locators.BUTTON_SIGN_IN)
+
+    def login(self, person):
+        self.clear_boxes()
+        self.enter_credentials(*Credentials[person])
+        self.click_sign_in()
+
+    def login_user(self,username, password):
```
Comment (TanyaKalistruk) [inline pages/auth_page.py:31]: this method and method `login` are the same

File: pages/header.py [hunk 3]
```diff
@@ -11,8 +14,26 @@ def __init__(self, driver):
...
+        self.click_element_by_text_simple(self.locators.LOG_OUT)
+
+    def click_log_in(self):
+        """Clicks on the round icon"""
+        self.click_element_by_text_simple(self.locators.LOG_IN)
+
+    def is_logined(self):
+        """ Check if user logined: if 'logout' button exist == logined """
+        self.click_icon()
+        # time.sleep(5) #==========================
```
Comment (TanyaKalistruk) [inline pages/header.py]: what does it mean? from this line to the end of the function?

File: tests/test_forgot_password.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+from config import EMAIL_SUBJECT_PASSW_RECOVERY, USERNAME_PASSW_RECOVERY, OLD_PASSWORD, NEW_PASSWORD
+from utilities.get_email import get_link
+from pages.forgot_password_page import ForgotPasswordPage
+from pages.confirm_password_page import ConfirmPassword
+
+
+class TestForgotPassword(BasePage):
+
+    def test_forgot_password(self):
+        header = Header(self.driver)
```
Comment (TanyaKalistruk) [inline tests/test_forgot_password.py:15]: please, make `__init__` in class TestForgotPassword and create objects header, login, etc and what you need for tests in this class and do not do it in tests

File: tests/test_forgot_password.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        confirmation_password.enter_confirm_password(NEW_PASSWORD)
+        confirmation_password.click_register_button()
+
+        assert login.is_password_restored()
+        login.click_ok()
+
+        login.login_user(USERNAME_PASSW_RECOVERY, OLD_PASSWORD)
+        assert header.is_logined()
+
+if __name__ == "__main__":
```
Comment (TanyaKalistruk) [inline tests/test_forgot_password.py]: please, delete this, you run tests with nosetest

File: tests/test_login_out.py [hunk 6]
```diff
@@ -1,23 +1,29 @@
...
 
-class TestTrial(BasePage):
+class TestLogin(BasePage):
 
-    def test_login_logout(self, person='USER'):
-        driver = self.driver
-        start = Header(driver)
-        signin = SignInPage(driver)
+    def test_login_logout(self):
+        header = Header(self.driver)
```
Comment (TanyaKalistruk) [inline tests/test_login_out.py:15]: the same thing with `__init__`

File: tests/test_login_out.py [hunk 7]
```diff
@@ -1,23 +1,29 @@
...
+        page.enter_sign_in_password(PASS)
+        page.click_sign_in()
 
-        signin.login(person)
+        assert header.is_logined()
 
-        start.transit('Profile')
-        start.transit('Log out')
 
 if __name__ == "__main__":
```
Comment (TanyaKalistruk) [inline tests/test_login_out.py]: delete

File: tests/test_sign_up.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+from pages.auth_page import AuthPage
+from pages.vacancies_page import VacanciesPage
+from config import EMAIL_SUBJECT_SIGNUP, USERNAME_SIGNUP, PASSWORD
+from utilities.get_email import get_link
+
+
+class TestSignUp(BasePage):
+
+    def test_sign_up(self):
+        header = Header(self.driver)
```
Comment (TanyaKalistruk) [inline tests/test_sign_up.py:13]: `__init__`

File: tests/test_sign_up.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+        assert vacancies.is_confirmation_sent()
+
+        link = get_link(EMAIL_SUBJECT_SIGNUP)
+        vacancies.click_confirmation_link(link)
+
+        page.login_user(USERNAME_SIGNUP, PASSWORD)
+        assert header.is_logined()
+
+if __name__ == "__main__":
```
Comment (TanyaKalistruk) [inline tests/test_sign_up.py]: delete

**Comment by Vilkoc** [comment]
this was mistake

**Pull request closed without merging**